azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -1212,20 +1233,23 @@ private void deletePod(final int executionId) throws ExecutorManagerException {
   private void deleteService(final int executionId) throws ExecutorManagerException {
     final String serviceName = getServiceName(executionId);
     try {
-      final V1Status deleteResult = this.coreV1Api.deleteNamespacedService(
-          serviceName,
-          this.namespace,
-          null,
-          null,
-          null,
-          null,
-          null,
-          new V1DeleteOptions());
-      logger.info("ExecId: {}, Action: Service Deletion, Service Name: {}, code: {}, message: {}",
-          executionId,
-          serviceName,
-          deleteResult.getCode(),
-          deleteResult.getMessage());
+      // Using GenericKubernetesApi due to a Known issue in K8s Java client and OpenAPIv2:
+      // See more here: https://github.com/kubernetes-client/java/wiki/6.-Known-Issues
+
+      GenericKubernetesApi<V1Service, V1ServiceList> serviceClient =
+          new GenericKubernetesApi<>(V1Service.class, V1ServiceList.class, "",
+              "v1", "services", this.client);
+      V1Service deletedService = serviceClient.delete(
+          this.namespace, serviceName).throwsApiException().getObject();
+      if (deletedService != null) {
+        logger.info(
```

Comment:
Is this the more common case? If not, then swap the if else blocks.

Reply:
ack